### PR TITLE
Formatting and documentation

### PR DESCRIPTION
**Pull request by curiousleo**
I left a couple of questions in comments in `normalizeBody`.

### CODE REVIEW

File: src/Language/Fixpoint/Defunctionalize.hs [hunk 1]
```diff
@@ -203,8 +203,3 @@ freshSym t = do
   let x = intSymbol "lambda_fun_" n
   modify $ \s -> s {dfFresh = n + 1, dfBinds = insertSEnv x t (dfBinds s)}
   return x
-
-
--- | getLams and getRedexes return the (previously seen) lambdas and redexes,
---   after "closing" them by quantifying out free vars corresponding to the
---   fresh binders in `dfBinds`.
```
Comment (curiousleo) [inline src/Language/Fixpoint/Defunctionalize.hs:210]: `getLams` and `getRedexes` no longer exist.

File: src/Language/Fixpoint/Solver/PLE.hs [hunk 2]
```diff
@@ -1154,42 +1161,36 @@ instance Normalizable Equation where
...
-      | f `elem` syms e
-      = go' e
-    go e
-      = e
-
-    go' (PAnd [PImp c e1,PImp (PNot c') e2])
-      | c == c' = EIte c e1 (go' e2)
-    go' e = e
-
-_splitBranches :: Symbol -> Expr -> [(Expr, Expr)]
```
Comment (curiousleo) [inline src/Language/Fixpoint/Solver/PLE.hs:1167]: `_splitBranches` was unused.

File: src/Language/Fixpoint/Solver/PLE.hs [hunk 3]
```diff
@@ -1154,42 +1161,36 @@ instance Normalizable Equation where
...
-_splitBranches f = go
-  where
-    go (PAnd es)
-      | f `elem` syms es
-      = go' <$> es
-    go e
-      = [(PTrue, e)]
+    -- Q: Why do we only simplify the expression if f is a free symbol in it?
```
Comment (curiousleo) [inline src/Language/Fixpoint/Solver/PLE.hs]: Question 1: Why do we only simplify the expression if f is a free symbol in it?

File: src/Language/Fixpoint/Solver/PLE.hs [hunk 4]
```diff
@@ -1154,42 +1161,36 @@ instance Normalizable Equation where
...
+    go e | f `elem` syms e = go' e
+    go e                   = e
 
-    go' (PImp c e) = (c, e)
-    go' e          = (PTrue, e)
+    -- | Simplifies a particular logical pattern:
+    -- (c => e1) /\ ((not c) => e2) --> if c then e1 else e2
+    --
+    -- Q: Why do we recurse into the right expression (only)?
```
Comment (curiousleo) [inline src/Language/Fixpoint/Solver/PLE.hs]: Question 2: Why do we recurse into the right expression (only)?

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 3 above)
Comment (nikivazou) [reply src/Language/Fixpoint/Solver/PLE.hs]: `normalizeBody f bd` is  only called when `bd` is the definition of `f`. We only need to normalize when `f` is recursive. So, we have this check for efficiency: I believe nothing bad will happen if you normalize anyways, but you do not need to.

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 4 above)
Comment (nikivazou) [reply src/Language/Fixpoint/Solver/PLE.hs]: Again, the expression has a concrete format (that came from Haskell's reflect annotations). The left expression is a conjuction of data constructor checkers, so you do not need to recurse there.

**Comment by curiousleo** [comment]
Thanks for the explanations @nikivazou! I've resolved the questions in the documentation comments.

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 3 above)
Comment (ranjitjhala) [reply src/Language/Fixpoint/Solver/PLE.hs]: Can we add @nikivazou 's remark to answer the question in the comment?

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 4 above)
Comment (ranjitjhala) [reply src/Language/Fixpoint/Solver/PLE.hs]: again, may be good to add @nikivazou 's answer into the comment?

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 3 above)
Comment (facundominguez) [reply src/Language/Fixpoint/Solver/PLE.hs]: I think this has be done in the last commit: eaff89e

File: src/Language/Fixpoint/Solver/PLE.hs (hunk 4 above)
Comment (facundominguez) [reply src/Language/Fixpoint/Solver/PLE.hs]: This has been added in eaff89e a well.

**Comment by facundominguez** [comment]
I merged #520, which I created to fix the conflicts in this branch. Please, let me know if anything looks amiss.

**Comment by ranjitjhala** [comment]
🙏🙏🙏

**Pull request closed without merging**